### ISSUE TITLE
client#query doesn't return updated data after performing a mutation

### ISSUE THREAD

**Issue by goje87** [apollostack/apollo-client#162]
For now I have a simple schema. And in connectors I used mongoose to deal with mongodb.

```
type User {
  _id: String!
  mobileNum: String,
  fullName: String
}

type RootQuery {
  user(mobileNum: String): User
}

type RootMutation {
  updateUser(mobileNum: String, fullName: String): User
}
```

At client side I simply do:

```javascript
var networkInterface = createNetworkInterface('http://localhost:8000', {
  credentials: 'include'
});

var client = new ApolloClient({
  networkInterface: networkInterface
});

var subs = client.watchQuery({query: `query { user(mobileNum: "${mobileNum}") { _id, mobileNum, fullName }}`})
  .subscribe({
    next: function(result) { console.log('next: ', result); },
    error: function(err) { console.err(err); } 
  });

```

Lets assume that initial user fetched has `fullName` as `Name One`. Now, if I run a mutation 

```javascript
client.mutate({
  mutation: `mutation {updateUser(mobileNum: "${mobileNum}", fullName: "Name Two") { fullName }}`
  })
.then((result) => { console.log(result) });
```

I see that `subs#next` is called twice. Both the time it prints `Name One` in the console. Then I find the response of the mutation call being printed with `fullName` as `Name Two`. Now, even if I call `client#query` on `user(mobileNum: "${mobileNum}")`, it returns `Name One`. But, if I call `subs.refetch()`, local store is updated. Both `query` and `watchQuery` are now printing `Name Two`.

Is there something that I'm doing wrong? Would appreciate your help in debugging this issue.

**Comment by stubailo** [comment]
Hey, right now the recommended pattern is to call `refetch` after the mutation. This is the result of a few related issues:

1. The callback is fired because we haven't implemented #124, so we sometimes fire unnecessary change notifications
2. We don't document well what happens after a mutation, which is also mentioned in #144 

I'm also working on documenting a better way to get refetches after a mutation, because right now that's not clear at all. I'm planning on documenting how to use `id` fields on your objects to incorporate the mutation result directly into the store.

Going to close this because I think the other issues cover the stuff you brought up - let me know if there is something I missed, and please post on there if you have thoughts about how these should be implemented!

**Comment by goje87** [comment]
Thanks @stubailo!! Sure.. would do so...
